pkg/registry/rbac/rest/storage_rbac.go

```diff
@@ -209,14 +233,15 @@ func (p *PolicyData) EnsureRBACPolicy() genericapiserver.PostStartHookFunc {
 			}
 
 			// ensure bootstrap roles are created or reconciled
+			clusterRoleService := newServiceTracker()
```

Comment:
this is inside the wait.Poll loop, so it forgets the service was available if the overall bootstrap has to retry